tests/test_pipelines/test_auto_augment.py

```diff
@@ -0,0 +1,115 @@
+        transform = dict(type='Translate', magnitude=0.5, direction='diagonal')
+        build_from_cfg(transform, PIPELINES)
+
+    # test assertion for invalid value of random_negative_prob
+    with pytest.raises(AssertionError):
+        transform = dict(
+            type='Translate', magnitude=0.5, random_negative_prob=100)
+        build_from_cfg(transform, PIPELINES)
+
+    # test case when magnitude=0, therefore no translate
+    results = construct_toy_data()
+    transform = dict(type='Translate', magnitude=0., prob=1.)
+    pipeline = build_from_cfg(transform, PIPELINES)
+    results = pipeline(results)
+    assert (results['img'] == results['ori_img']).all()
+
+    # test case when prob=0, therefore no shear
+    results = construct_toy_data()
+    transform = dict(type='Translate', magnitude=1., prob=0.)
+    pipeline = build_from_cfg(transform, PIPELINES)
+    results = pipeline(results)
+    assert (results['img'] == results['ori_img']).all()
+
+    # test translate horizontally, magnitude=0.5
+    results = construct_toy_data()
+    transform = dict(
+        type='Translate',
+        magnitude=0.5,
+        pad_val=0,
+        prob=1.,
+        random_negative_prob=0.)
+    pipeline = build_from_cfg(transform, PIPELINES)
+    results = pipeline(results)
+    translated_img = np.array([[0, 0, 1, 2], [0, 0, 5, 6], [0, 0, 9, 10]],
+                              dtype=np.uint8)
+    translated_img = np.stack([translated_img, translated_img, translated_img],
+                              axis=-1)
+    assert (results['img'] == translated_img).all()
+    assert (results['img'] == results['img2']).all()
+
+    # test translate vertically, magnitude=-0.5
+    results = construct_toy_data()
+    transform = dict(
+        type='Translate',
+        magnitude=-0.5,
+        pad_val=0,
+        prob=1.,
+        direction='vertical',
+        random_negative_prob=0.)
+    pipeline = build_from_cfg(transform, PIPELINES)
+    results = pipeline(results)
+    translated_img = np.array([[9, 10, 11, 12], [0, 0, 0, 0], [0, 0, 0, 0]],
+                              dtype=np.uint8)
+    translated_img = np.stack([translated_img, translated_img, translated_img],
+                              axis=-1)
+    assert (results['img'] == translated_img).all()
+
+    # test translate vertically, magnitude=0.5, random_negative_prob=1
+    results = construct_toy_data()
+    transform = dict(
+        type='Translate',
+        magnitude=0.5,
+        pad_val=0,
+        prob=1.,
+        direction='vertical',
+        random_negative_prob=1.)
+    pipeline = build_from_cfg(transform, PIPELINES)
+    results = pipeline(results)
+    translated_img = np.array([[9, 10, 11, 12], [0, 0, 0, 0], [0, 0, 0, 0]],
+                              dtype=np.uint8)
+    translated_img = np.stack([translated_img, translated_img, translated_img],
+                              axis=-1)
+    assert (results['img'] == translated_img).all()
```

Comment:
Also, please check the test cases: https://github.com/open-mmlab/mmcv/blob/master/tests/test_image/test_geometric.py#L488